### PR TITLE
CIV-14228 GA Upload additional documents

### PR DESCRIPTION
**Pull request by jeswanth-hmcts**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-14228

### Change description ###

GA Upload additional documents
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by paul-pearson — DISMISSED** [review]

File: src/main/modules/i18n/locales/cy.json [hunk 1]
```diff
@@ -2575,6 +2575,16 @@
         "WHY_NOT_ACCEPT": "Why do you not accept the defendant's offer?",
         "WHY_NOT_ACCEPT_2": "Why you do not accept the defendant's offer?"
       },
+      "ADDITIONAL_DOCUMENTS": {
+        "ADDITIONAL_DOCUMENTS_CAPTION": "Upload additional documents",
+        "UPLOAD_DOCUMENTS_TITLE": "Upload documents",
+        "ADDITIONAL_DOCUMENTS_ROW1": "You should only upload documents that are relevant to your application, for example if the judge has ordered a hearing and instructed you to provide documents ahead of this.",
```
Comment (manish14836) [inline src/main/modules/i18n/locales/cy.json:2581]: Sort these

File: src/main/routes/features/generalApplication/additionalDocuments/checkAnswersController.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+gaAdditionalDocCheckAnswerController.post(GA_UPLOAD_ADDITIONAL_DOCUMENTS_CYA_URL, (async (req: AppRequest, res: Response, next: NextFunction) => {
+  try {
+    const { gaId, id: claimId } = req.params;
+    const claim = await getClaimDetailsById(req);
+    const uploadedDocuments = prepareCCDData(claim.generalApplication.uploadAdditionalDocuments);
+    const generalApplication = {
+      uploadDocument: uploadedDocuments,
+    };
+    await gaServiceClient.submitEvent(ApplicationEvent.UPLOAD_ADDL_DOCUMENTS, gaId, generalApplication as unknown, req);
+    res.redirect(GA_UPLOAD_ADDITIONAL_DOCUMENTS_SUBMITTED_URL.replace(':id', claimId).replace(':gaId', gaId));
```
Comment (manish14836) [inline src/main/routes/features/generalApplication/additionalDocuments/checkAnswersController.ts:39]: Suggestions : Create method for this which can be use in other future controllers.

File: src/main/routes/features/generalApplication/additionalDocuments/uploadAdditionalDocumentsController.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+uploadAdditionalDocumentsController.get(GA_UPLOAD_ADDITIONAL_DOCUMENTS_URL, (async (req: AppRequest, res: Response, next: NextFunction) => {
+  try {
+    const { gaId, id } = req.params;
+    const uploadedDocument = new UploadAdditionalDocument();
+    let form = new GenericForm(uploadedDocument);
+    const redisKey = generateRedisKey(req);
+    const claim = await getClaimDetailsById(req);
+    const gaDetails = claim.generalApplication;
+
+    if (req?.session?.fileUpload) {
```
Comment (manish14836) [inline src/main/routes/features/generalApplication/additionalDocuments/uploadAdditionalDocumentsController.ts:31]: Do we need Option for request here?

File: src/main/modules/i18n/locales/cy.json (hunk 1 above)
Comment (jeswanth-hmcts) [reply src/main/modules/i18n/locales/cy.json:2580]: sorted has suggested

File: src/main/routes/features/generalApplication/additionalDocuments/checkAnswersController.ts (hunk 2 above)
Comment (jeswanth-hmcts) [reply src/main/routes/features/generalApplication/additionalDocuments/checkAnswersController.ts]: chnaged to appid

File: src/main/routes/features/generalApplication/additionalDocuments/uploadAdditionalDocumentsController.ts (hunk 3 above)
Comment (jeswanth-hmcts) [reply src/main/routes/features/generalApplication/additionalDocuments/uploadAdditionalDocumentsController.ts]: removed ? after req

**Review by manish14836 — DISMISSED** [review]

**Review by paul-pearson — DISMISSED** [review]

**Pull request merged**